src/django_components/dependencies.py

```diff
@@ -112,23 +111,29 @@ def _hash_comp_cls(comp_cls: Type["Component"]) -> str:
 def _gen_cache_key(
     comp_cls_hash: str,
     script_type: ScriptType,
+    input_hash: Optional[str],
```

Comment:
Many of the functions in this file now accept additional `input_hash` argument, which refers to a hash of the serialized JS or CSS variables.

So e.g. if I defined CSS variables like so:
```py
def get_css_data(self):
    return {"border_width": "3px", "button_color": "blue" }
```

Then this would be serialized into JSON as:
```json
{
  "border_width": "3px",
  "button_color": "blue"
}
```

And since JSON is a string, the input hash would be derived from the JSON as:
```py
input_hash = md5(json_data.encode()).hexdigest()[0:6]
```

So the first 6 characters of the MD5 hash.

This is then passed around as the ID that points to the file that holds the CSS / JS variables.

E.g. If I had a component `Table`, and I generated the CSS variables above, and I got hash `a1b2c3`, then I would know that:

1. An endpoint like `/components/cache/table.css` would fetch the CSS from `Table.css`
2. An endpoint like `/components/cache/table.a1b2c3.css` would fetch the CSS variables for component `Table` as described above.